### ISSUE TITLE
Adafruit CLUE Radio Communications with micropython radio module on micro:bit v2

### ISSUE THREAD

**Issue by burksfamly** [adafruit/Adafruit_CircuitPython_CLUE#62]
I have attempted to initiate discussion on Adafruit forum without success...so I am trying here.

I have developed a mesh sensor network of micro:bits that is quite useful for my needs.
I wrote the Tx/Rx packet protocol and error recovery using the micopython radio module 
for the micro:bit. I am very pleased with the outcome (e.g., packet size, range, error recovery).

During my research, I found that the micropython radio module for the micro:bit v2 is an 
island and is not compatible with anything else, unless you use the micropython radio module
on another micro:bit (which all nodes, repeater, and control nodes in my mesh are written in
this environment). Even using the MakeCode radio module on the micro:bit v2, one is unable to 
communicate with the micropython radio module on the micro:bit v2.

I have a question for some contributor that can shed light on what I would like to explore,
as I am not able to get a response sufficient to make a decision if I want to proceeed and 
how deep the rabbit hole will go (i.e., think time committment).

I would like to insert an Adafruit CLUE into my mesh sensor network and have it communicate
to the micropython radio module on the micro:bit v2. I do NOT want to use another type of
radio configuration (BLE) nor do I want to use a serial USB cable connection. My existing
architecture is loosely as follows:

mbSensorNode(s)<-radio->mbRptrNode<-radio->mbControlNode<-USB->RPi<->rVNC/TwilioSMS<->myPhone

My issue in the above configuration is the "<-radio->" link for the Adafruit CLUE.

What I believe I need is a library (etiher Arduino IDE C++ or CircuitPython, don't care
which one) that I can call functions from a sketch housed on the Adefruit CLUE to setup 
the Adafruit CLUE nRF chip, configure, send, recieve and parse packets (structure of bytes 
based on my mesh design) that are visible when polled by a sketch housed on the micropython 
radio module micro:bit v2 environment.

At this point I am CLUEless :) on how to approach this. I understand an attack to achieve this
would loosely be as follows:

1. Open the hood on the micropython radio module for micro:bit v2 to determine exactly how
   they have written code to setup micro:bit nRF chip and Tx/Rx data.

2. Research the Adafruit CLUE nRF chip to ensure that it can and how to setup for comms
   with micropython radio module on the micro:bit v2.

3. Select either Arduino IDE C++ or CircuitPython and attempt to develop a library
   that is compatible with the micropython radio module on the micro:bit v2. And/Or
   use the Nordic SDK as mentioned from another forum respondor. 

I am not clear on the following:

1. Has this been done?

2. Can this be done?

3. What exactly (in detail) would I need to setup as a development environment, that
   would avoid pitfalls that I am not able to see at this time. What would be the 
   best approach?

Could I do this? Yes, but it may end up taking me 6 months. Should I do this? That
is what I am trying to measure. For example, it is whelming to setup and understanding
a development environment to develop a simple Android Phone Application. It is also
whelming to setup an evironment for developing a 4-stack application.

Any guidance or suggestions would be greatly appreciated. Also, if someone has a
deep understanding of the development environment that would be necessary, I 
would be interested in collaborating.

Thank you.

**Comment by burksfamly** [comment]
tannewt,

Thank you SO much for your reply and I will dive deeper into your response in the next day or two.
In the interim, I caught in your response the phrase "I decided against..." so I must be talking to the right person.
About three years ago I experimented with many microcontrollers and bluetooth for a mesh solution. During that
development, I had NOTHING except mare problems...it always just met minimum criteria, but then something else
came up. It was good if I needed to be in close proximity to my laptop or RPi, but for mesh communications within
a home (three stories, from basement to second floor) it just did not work.

I got a freebee from Adafruit one day...the micro:bit v2...I experimented and within several months....I had a full blown
mesh (with auto routing requests and error correction for dropped packets)....I had to eventually pull that functionality
due to sketch space, but now it works very reliably....also the buffer size for radio comms is huge...I don't even have to
do bit encoding to reduce packet size, I have about a dozen ASCII packet elements in the packet header with approximately
a 64-byte payload buffer (which I don't really need to use. So the micropython import radio module for micro:bit v2 is
WAY more powerful than I expected...and is very easy to modify and maintain as a mesh sensor network...I am beside
myself...This is the reason why I want to explore integrating the CLUE with my network.

You are correct...on the micro:bit the micro:python radio module is mutually exclusive with BLE...IIRC...BLE is NOT fully
implemented on the micro:bit. 

For my needs, I do not need BLE...but I do need to be able to communicate with the micro:bit as if it was using the micropython radio module on a micro:bit v2.

The learning curve or if you will the time to deconstruct and rebuild may not be worth it for me. However, working with you, such an effort may make sense. 

Don't get me wrong, BLE has it's place, but my application (or similar ones) are not solve by BLE based on my tests and tinkering.

What would it take to develop such a compatible library module for the CLUE, even if it means to forego use of BLE on the CLUE if one used such a module.

I don't know for sure, but my gut tells me that others would benefit from this too.

Thank you.
